examples/claimAndSwap.ts

```diff
@@ -0,0 +1,188 @@
+      ABI_CODER.encode(
+        ["address", "address", "uint256", "uint256"],
+        [token, receiver, amount, vesting_duration]
+      )
+    ),
+  });
+  const receipt = await tx.wait();
+  if (receipt?.status === 0)
+    throw new Error("Vesting contract creation failed");
+
+  const vestingContractAddr = getAddress(
+    hexDataSlice(receipt?.logs[0].topics[2] as string, 12)
+  );
+  console.log(`vesting contract created at ${vestingContractAddr}`);
+  return vestingContractAddr as string;
+};
+
+const claimAndSwap: Parameters<typeof withAnvilProvider>[0] = async (
+  provider: ethers.JsonRpcProvider,
+  signers: ethers.Wallet[],
+  { factory, implementation, proxyInitCode }
+) => {
+  const funder = signers[0];
+  const user = signers[1];
+  const funderAddr = await funder.getAddress();
+  const userAddr = await user.getAddress();
+  const amount = parseEther("1");
+
+  await wrapEther(provider, funderAddr, amount);
+
+  const vestingContractAddr = await createVest(
+    provider,
+    funderAddr,
+    WETH,
+    userAddr,
+    amount,
+    500n
+  );
+
+  await provider.send("evm_increaseTime", [501]);
+
+  // compute the proxy address with CowShedSdk
+  const shedSdk = new CowShedSdk({
+    factoryAddress: factory,
+    implementationAddress: implementation,
+    chainId: 1,
+    proxyCreationCode: proxyInitCode,
+  });
+
+  const validTo = Math.floor(new Date().getTime() / 1000) + 7200;
+
+  const order: Order = {
+    sellToken: WETH,
+    buyToken: COW,
+    receiver: userAddr,
+    sellAmount: amount,
+    buyAmount: parseEther("1000"),
+    validTo,
+    appData: "",
+    feeAmount: 0n,
+    kind: OrderKind.SELL,
+    partiallyFillable: true,
+    sellTokenBalance: OrderBalance.ERC20,
+    buyTokenBalance: OrderBalance.ERC20,
+  };
+
+  // pre-hooks
+  const calls: ICall[] = [
+    // approve the bridge to spend the swapped usdc
+    {
+      target: vestingContractAddr,
+      callData: fnCalldata(
+        "claim(address)",
+        ABI_CODER.encode(["address"], [userAddr])
+      ),
+      value: 0n,
+      isDelegateCall: false,
+      allowFailure: false,
+    },
+  ];
+  const nonce = ethers.encodeBytes32String("first");
+
+  // signing the hooks intent
+  const hashToSign = shedSdk.hashToSignWithUser(
+    calls,
+    nonce,
+    BigInt(validTo),
+    userAddr
+  );
+  console.log("hash to sign", hashToSign);
+  const signature = user.signingKey.sign(hashToSign);
+  console.log("actual signature", signature.r, signature.s, signature.v);
+  const encodedSignature = CowShedSdk.encodeEOASignature(
+    BigInt(signature.r),
+    BigInt(signature.s),
+    signature.v
+  );
+
+  const hooksCalldata = CowShedSdk.encodeExecuteHooksForFactory(
+    calls,
+    nonce,
+    BigInt(validTo),
+    userAddr,
+    encodedSignature
+  );
+
+  const hooks = {
+    pre: [
+      {
+        target: factory,
+        callData: hooksCalldata,
+        gasLimit: "9999999999999999999999",
```

Comment:
Being overzealous like this with gas limits may result in the API refusing the order (this gas limit is beyond block limits - but also likely beyond the ETH required when simulating settlement).